assets/src/sass/utilities/_theme-supports.scss

```
@@ -39,6 +39,56 @@ $color-palette-reverse:
 	}
 }
 
+// Support accessible color combinations for light background colors.
+$color-palette-reverse-light:
+	"orange",
+	"yellow",
+	"pink",
+	"bright-blue",
+	"bright-green",
+	"bright-yellow",
+	"light-pink",
+	"light-purple",
+	"light-bright-blue",
+	"light-bright-green",
+	"lightest-pink",
+	"mid-bright-yellow",
+	"light-bright-yellow",
+	"lightest-yellow",
+	"lightest-purple",
+	"lightest-blue",
+	"lightest-green",
+	"base",
+	"black-50",
+	"black-25",
+	"black-10",
+	"black-05",
+	"border-light",
+	"border-base";
+
+@each $color in $color-palette-reverse-light {
+	.has-#{$color}-background-color {
+		color: var(--wp--preset--color--main) !important;
+
+		.wp-block-shiro-button.is-style-as-link::before,
+		.wp-block-shiro-button.is-style-as-arrow-link::before,
+		.is-style-transparent.has-icon .wp-block-button__link:not(:active)::before,
+		.is-style-as-link.has-icon .wp-block-button__link::before,
+		.is-style-as-arrow-link.has-icon .wp-block-button__link::before {
+			filter: brightness(0) invert(1) !important;
```

```suggestion
			filter: brightness(0)!important;
```

We want the icons to be black, like the text.